transport/transport.go

```diff
@@ -559,12 +559,6 @@ func wait(ctx context.Context, done, goAway, closing <-chan struct{}, proceed <-
 	case <-closing:
 		return 0, ErrConnClosing
 	case i := <-proceed:
```

Comment:
The caller of this function will not reset channel proceed so proceed will not be readable by other channels.
Is this the reason?